### ISSUE TITLE
shellcheck should throw an error when a bash-specific parameter expansion (${v,,} , ${v^^}, etc.) is used in a `dash` script.

### ISSUE THREAD

**Issue by emanuele6** [koalaman/shellcheck#2013]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

day="${1-$(date +%A)}"

case "${day,,}" in
    sunday)    echo '日' ;;
    monday)    echo '月' ;;
    tuesday)   echo '火' ;;
    wednesday) echo '水' ;;
    thursday)  echo '木' ;;
    friday)    echo '金' ;;
    saturday)  echo '土' ;;
    *)         exit 1    ;;
esac
```

#### Here's what shellcheck currently says:
```
```

#### Here's what I wanted or expected to see:
```
case "${day,,}" in
      ^------^ SC****: In POSIX sh, ${var,,} is not defined.
```

**Comment by emanuele6** [comment]
This should be an "error": at least if the shell being checked is `dash` since the script will fail with an error:
```shell
$ shellcheck --shell=dash weekdaykanji; echo "$?"
0
$ dash weekdaykanji
weekdaykanji: 5: Bad substitution
```

**Comment by emanuele6** [comment]
This was "fixed" by 5b86777 @koalaman . It is not reported as a warning both for `POSIX sh` and `dash`.

But this should really be an error, not a warning (at least for `dash`)!

**Comment by koalaman** [comment]
I agree, these simply won't work in dash and have therefore upgraded from warnings to errors. Thanks!
